extension/src/popup/views/RecoverAccount/index.tsx

```diff
@@ -26,41 +31,90 @@ import { PasswordRequirements } from "popup/components/PasswordRequirements";
 
 import "./styles.scss";
 
-export const RecoverAccount = () => {
-  const publicKey = useSelector(publicKeySelector);
-  const authError = useSelector(authErrorSelector);
-  const publicKeyRef = useRef(publicKey);
+interface PhraseInputProps {
+  phraseInput: string;
+  index: number;
+  isDisabled: boolean;
+  handleMnemonicInputChange: (value: string, index: number) => void;
+}
+
+const PhraseInput = ({
+  phraseInput,
+  index,
+  isDisabled,
+  handleMnemonicInputChange,
+}: PhraseInputProps) => {
+  const [isTextShowing, setIsTextShowing] = useState(false);
+  const [inputValue, setInputValue] = useState("");
 
+  return (
+    <div
+      key={phraseInput}
+      className={`RecoverAccount__phrase-input ${
+        isDisabled ? "RecoverAccount__phrase-input--disabled" : ""
+      }`}
```

Comment:
the strategy here is to create the max number of inputs possible for a mnemonic phrase (in this case, 24) and then just show/hide the needed fields depending on how many the user selects